lib/pavilion/plugins/commands/status.py

```diff
@@ -139,6 +140,83 @@ def get_tests(pav_cfg, args, errfile):
     return test_list
 
 
+def status_via_time(pav_cfg, args, errfile):
+    """Status via time grabs tests based on the given time argument
+    It will check tests datetime objects and returns all tests up to
+    a certain time. Example. 'pav status --time 5 hours' will display
+    all test run up to 5 hours ago.
+    :param pav_cfg: The pavilion config.
+    :param args: Tests via command line arguments
+    :param errfile: Stream to output errors as needed
+    :returns: List of dictionary objects"""
+    # Converting time arguments into valid datetime object.
+    try:
+        time_amount = int(args.time[0])
+        time_unit = args.time[1]
+    except ValueError:
+        output.fprint("Invalid Format. Use the from: $AMOUNT $UNIT e.g."
+                      " '5 hours'", file=errfile, color=output.RED)
+        return []  # No tests to return.
+
+    # Convert time into hours
+    # You lose some accuracy if you do like, `-t 15 years`, but
+    # I hope nobody will have a 15 year old pav directory.
+    if time_unit == 'second' or time_unit == 'seconds':
+        time_amount = time_amount/3600
+    elif time_unit == 'minute' or time_unit == 'minutes':
+        time_amount = time_amount/60
+    elif time_unit == 'hour' or time_unit == 'hours':
+        time_amount = time_amount * 1
+    elif time_unit == 'day' or time_unit == 'days':
+        time_amount = time_amount * 24
+    elif time_unit == 'week' or time_unit == 'weeks':
+        time_amount = time_amount * 168
+    elif time_unit == 'month' or time_unit == 'months':
+        time_amount = time_amount * 720
+    elif time_unit == 'year' or time_unit == 'years':
+        time_amount = time_amount * 8760
+    else:
+        output.fprint("Invalid time unit, --time only accepts "
+                      "second(s), minute(s), hour(s), day(s), "
+                      "week(s), month(s), and year(s).",
+                      file=errfile, color=output.RED)
+        return []  # No tests to return.
+
+    try:
+        search_date = datetime.today() - timedelta(hours=time_amount)
+    except OverflowError:
```

Comment:
This isn't quite true. Datetime can't go back before year 1 AD.

Reply:
I already catch that.